### PR TITLE
[Table]: Add information under table header about how many rows were selected as well as provide options for bulk actions

### LINKED ISSUE #9864: [Feature Request] Table multiple select with actions
### Existing Component
Yes

### Component Name
Table

### Description
Currently, there's no way to perform bulk actions on table row selections although selections can be made. If something showing bulk actions popped out somewhere, it would be helpful to many use cases. :)

### PR DESCRIPTION
**Pull request by smilee**
Fixes #9864 

You can now add a slot to show how many rows have been selected as well as if you have any bulk actions available! 🐑

### CODE REVIEW

File: packages/table/src/table-header.js [hunk 1]
```diff
@@ -74,6 +74,8 @@ export default {
     // 是否拥有多级表头
     const isGroup = columnRows.length > 1;
     if (isGroup) this.$parent.isGroup = true;
+    const hasSelection = this.store.states.selection.length > 0;
+    hasSelection ? this.$parent.hasSelection = true : this.$parent.hasSelection = false;
```
Comment (QingWei-Li) [inline packages/table/src/table-header.js]: `this.$parent.hasSelection = hasSelection`

File: packages/table/src/table.vue [hunk 2]
```diff
@@ -304,6 +317,8 @@
 
       defaultSort: Object,
 
+      multipleSelect: Object,
```
Comment (QingWei-Li) [inline packages/table/src/table.vue]: remove it

File: packages/theme-chalk/src/table.scss [hunk 3]
```diff
@@ -246,6 +246,24 @@
     }
   }
 
+  .actions {
```
Comment (QingWei-Li) [inline packages/theme-chalk/src/table.scss]: remove it

File: packages/table/src/table.vue [hunk 4]
```diff
@@ -24,11 +24,18 @@
...
+        :multiple-select="multipleSelect"
         :style="{
           width: layout.bodyWidth ? layout.bodyWidth + 'px' : ''
         }">
       </table-header>
     </div>
+    <div
+      v-if="hasSelection"
+      class="el-table__bulk-wrapper"
+      ref="bulkWrapper">
```
Comment (QingWei-Li) [inline packages/table/src/table.vue]: remove it

File: packages/table/src/table.vue [hunk 5]
```diff
@@ -102,6 +109,12 @@
             width: layout.fixedWidth ? layout.fixedWidth + 'px' : ''
           }"></table-header>
       </div>
+      <div
+        v-if="hasSelection"
+        class="el-table__bulk-wrapper"
+        ref="bulkWrapper">
```
Comment (QingWei-Li) [inline packages/table/src/table.vue]: remove it

File: packages/table/src/table-header.js [hunk 6]
```diff
@@ -474,6 +476,10 @@ export default {
       document.body.style.cursor = '';
     },
 
+    handleActionClick(event, action, selection) {
```
Comment (QingWei-Li) [inline packages/table/src/table-header.js]: remove it

**Comment by smilee** [comment]
@QingWei-Li Thank you for your kind feedback and consideration. I need this slot for the design to work for my project. Being able to interact with the bulk actions in this manner improves the user experience. This feature is more about the design and less about the functionality. You are right when you said that the functionality already exists. Thank you again for considering this slot to make your amazing project more useful for designers like me.

File: packages/table/src/table.vue [hunk 7]
```diff
@@ -24,11 +24,13 @@
         :store="store"
         :border="border"
         :default-sort="defaultSort"
+        :multiple-select="multipleSelect"
```
Comment (QingWei-Li) [inline packages/table/src/table.vue]: Please remove it

**Comment by Leopoldthecoder** [comment]
Sorry, this doesn't seem to be a common use case, so we'll not be supporting this.

If you do need this slot, perhaps you can fork this library, do your customization, and publish it on npm.

Finally, this is a workaround of this feature: https://jsfiddle.net/23fv0f8c/11/ . Hacky, but works. Your decision.

Thanks for contributing.

**Pull request closed without merging**